core/domain/collection_services.py

```diff
@@ -247,6 +248,84 @@ def is_collection_summary_editable(collection_summary, user_id=None):
         or collection_summary.community_owned)
 
 
+def get_learner_collection_dict_by_id(
+        collection_id, user_id, strict=True, allow_invalid_explorations=False,
+        version=None):
+    """Creates and returns a dictionary representation of a collection given by
+    the provided collection ID. This dictionary contains extra information
+    along with the dict returned by collection_domain.Collection.to_dict()
+    which includes useful data for the collection learner view. The information
+    includes progress in the collection, information about explorations
+    referenced within the collection, and a slightly nicer data structure for
+    frontend work.
+
+    This raises a ValidationError if the collection retrieved using the given ID
+    references non-existent explorations.
+    """
+    collection = get_collection_by_id(
+        collection_id, strict=strict, version=version)
+
+    exp_ids = collection.exploration_ids
+    exp_summary_dicts = (
+        summary_services.get_displayable_exp_summary_dicts_matching_ids(
+            exp_ids))
+    exp_summaries_dict_map = {
+        exp_summary_dict['id']: exp_summary_dict
+        for exp_summary_dict in exp_summary_dicts
+    }
+
+    # TODO(#1461): Users should not be recommended explorations they
```

Comment:
Suggested format:

    # TODO(bhenning): Users should ... collection (see #1461).